examples/protocol_specification_ex/oef_search.yaml

```diff
@@ -37,3 +37,13 @@ ct:OefErrorOperation: |
       }
   OefErrorEnum oef_error = 1;
 ...
+---
+reply:
+  register_service: [oef_error]
+  unregister_service: [oef_error]
+  search_services: [search_result,oef_error]
+  search_result: []
+  oef_error: []
+roles: {agent, oef_node}
+end_states: [open, closed]
```

Comment:
I think this should be `successful, failed`. This dialogue seems clearly structured. An `oef_search` dialogue terminates with either `search_result` or `oef_error` message. In the former case, it is `successful`, in the latter `failed`.

Reply:
In case of registration it is neither successful nor failed as we never get a response...happy to have `end_states: [successful, failed]` though

Reply:
Again, if we want to avoid multiple "dialogue sequences" under this dialogue definition, we can make `unregister` a reply to `register`. This makes it a longer running interaction between oef and agent. That's if `unregister` always follows `register`.